cumulusci/tasks/metadeploy.py

```diff
@@ -83,12 +84,19 @@ def _init_task(self):
         Path(self.labels_path).mkdir(parents=True, exist_ok=True)
 
         if plan_name := self.options.get("plan"):
+            if self.project_config.lookup(f"plans__{plan_name}") is None:
+                raise TaskOptionsError(
+                    f"Plan {plan_name} not found in project configuration"
+                )
             plan_configs = {
                 plan_name: self.project_config.lookup(f"plans__{plan_name}")
             }
             self.plan_configs = plan_configs
         else:
             self.plan_configs = self.project_config.plans
+        # Handled exception for no plan
+        if self.plan_configs is None or len(self.plan_configs) == 0:
+            raise error("No plan found to publish in project configuration")
```

Comment:
I don't think you want this `error` symbol. You can raise a `CumulusCIException` here.